packages/x-data-grid/src/hooks/core/useGridProps.ts

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import type { RefObject } from '@mui/x-internals/types';
+import type { DataGridProps } from '../../models/props/DataGridProps';
+import type { GridPrivateApiCommon } from '../../models/api/gridApiCommon';
+import type { GridStateInitializer } from '../utils/useGridInitializeState';
+import type { GridStateCommunity } from '../../models/gridStateCommunity';
+import type { GridRowModel } from '../../models/gridRows';
+import { createSelector } from '../../utils/createSelector';
+import { GRID_ID_AUTOGENERATED } from '../features/rows/gridRowsUtils';
+
+type Props = Pick<DataGridProps, 'getRowId'>;
+
+const gridPropsStateSelector = (state: GridStateCommunity) => state.props;
+
+export const gridRowIdSelector = createSelector(
+  gridPropsStateSelector,
+  (props, row: GridRowModel) => {
+    if (GRID_ID_AUTOGENERATED in row) {
+      return row[GRID_ID_AUTOGENERATED];
+    }
+    return props.getRowId ? props.getRowId(row) : row.id;
+  },
+);
```

Comment:
In case this needs high performance, it could make sense to write it without the `gridPropsStateSelector` additional call.

Reply:
You mean not even wrapping with `createSelector`?

Reply:
Yes, and yes that would also be valid for other selectors, it's just that `getRowId` is quite fundamental so it's probably used more than other selectors.

Reply:
Sounds good 👍